shell/artifacts/Common/source/ShowCollection.js

```diff
@@ -135,7 +135,7 @@ recipe
 
       this._updateRenderModel(propsItems, this._state.hostedItems || []);
 
-      if (propsItems.length == 0 && !this._state.template) {
+      if (propsItems.length > 0 && !this._state.template) {
```

Comment:
maybe just 
`if (!this._state.template) {`
?
i don't remember from the top of my head why I added `propsItems.length` in the first place, but it was working for the gift shopping demo, and won't be working with this change.

also, note to self, I need to remember to port this change when/if Scott approves https://github.com/PolymerLabs/arcs/pull/941.

Reply:
It only works for me with `>0`. My recipe:

```
recipe
  create #wishlist as wishlist
  map #shortlist as shortlist
  Chooser
    choices <- shortlist
    resultList = wishlist

    ShowCollection
      collection <- wishlist
      hostedParticle = ShowProduct
```

As I recall I saw some different behavior with `create` vs `map` with regards to this.

It seems as though we're using `_setState({template...})` to control rendering flow, is that right? Is there still a `shouldRender()` or equivalent that might be more direct?